R/spread-process.R

```diff
@@ -165,32 +172,34 @@ setMethod("spread",
                                             pairs=FALSE))
               }
               #browser()
-
-
+              
+              
               #if there is only one potential, R converts this to a vector, instead of a matrix.
               # Force it back to a matrix
-#              if(length(potentials)==2) {
-#                potentials <- matrix(potentials,ncol=2)
-#              }
-
+              #              if(length(potentials)==2) {
+              #                potentials <- matrix(potentials,ncol=2)
+              #              }
+              
               # drop those ineligible
               #              if (!is.null(mask))
               #                potentials <- matrix(potentials[potentials[,2] %in% masked,], ncol=2)
-
+              
               # only accept cells that have no fire yet
               # Need to call matrix because of the cast where there is only one cell
               #potentials <- matrix(potentials[spreads[potentials[,2]]==0,], ncol=2)
               potentials <- potentials[spreads[potentials[,2]]==0,,drop=FALSE]
-
+              
               # If one pixels is selected as potential by more than one source
               #  Remove the duplication, and reorder the potentials so that it is not
               #  always the "first one", i.e., closest to top left of map, that is kept.
-              if(nrow(potentials)>0) {
-                potentials <- potentials[sample.int(nrow(potentials)),,drop=FALSE]
-              }
-              potentials <- potentials[!duplicated(potentials[,2]),,drop=FALSE]
-
-
+              if(NROW(potentials)>0) {
+                if(anyDuplicated(potentials[,2])){
+                  ## For testing purposes (to benchmark)
```

Comment:
It's necessary code you are right. I need to benchmark that part to see if it's worth doing a check every time to avoid a sample.int() or let the code as it was. I need to find the time to do it.

The change is made so that we will keep it if the speed actually improve overall.